### DISCUSSION TITLE
Override PenHelper failed error behavior or reporting

### DISCUSSION THREAD

**Discussion by emphasis87** [Excel-DNA/ExcelDna#692] [Q&A] [answered]
Is there a way to stop ExcelDna reporting on "IsInFormulaEditMode - PenHelper failed" ? Not sure how it happens, but I am queueing a lot of macro delegates though `ExcelAsyncUtil.QueueAsMacro` and this often happens. I don't want to show it to the users and block them. I would prefer to log it, or if possible decide what to do - retry, drop the delegate and so on.

**Comment by govert** [comment, marked as answer]
The error you see is an unexpected error - I would be interested if you have a reproducible way to get that error to pop up that you can share.
What is the exact Excel-DNA version you are using? We did change some of the relevant implementation (I think in version 1.7) to try to make it more robust.

You can suppress the default logging behaviour (which is to pop up the Log Display dialog for any error) by adding a .config file to your add-in:

```xml
<configuration>
 <system.diagnostics>
    <sources>
      <source name="ExcelDna.Integration" switchValue="Off">
      </source>
    </sources>
  </system.diagnostics>
</configuration>
```

But I'd prefer to figure out why that call is failing in your context.

**Reply by emphasis87** [reply]
Hi,
unfortunately I can not share this document. As far as I could determine, the error appears when they close the document, the window with the error pops up and the error repeats ad infinitum. Not sure if it is an issue of not disposing/unsubscirbing, but when they close the window it reappears and they need to kill the process
ExcelDna version is 1.7.0.

**Reply by govert** [reply]
Is it only happening with one specific document?
Does it have interesting VBA in, or anything special you can think of?

**Comment by govert** [comment]
@emphasis87 
I've changed the logging level from Error to Warn in version 1.8.0-rc.
Please have a look at whether this fixes your issue.
